Source/Core/Core/HW/MMIO.cpp

```diff
@@ -195,6 +195,105 @@ WriteHandlingMethod<T>* InvalidWrite()
   });
 }
 
+// Debug: specialization of the complex handling type which acts as wrapper for all other handler.
+// It will additionally show each memory access.
+template <typename T>
+class DebugHandlingMethod : public ReadHandlingMethod<T>,
+                            public WriteHandlingMethod<T>,
+                            ReadHandlingMethodVisitor<T>,
+                            WriteHandlingMethodVisitor<T>
+{
+public:
+  explicit DebugHandlingMethod(ReadHandlingMethod<T>* read_handler) : read_handler_(read_handler)
+  {
+    read_handler_->AcceptReadVisitor(*this);
+  }
+
+  explicit DebugHandlingMethod(WriteHandlingMethod<T>* write_handler)
+      : write_handler_(write_handler)
+  {
+    write_handler_->AcceptWriteVisitor(*this);
+  }
+
+  virtual ~DebugHandlingMethod() {}
+  void AcceptReadVisitor(ReadHandlingMethodVisitor<T>& v) const override
+  {
+    v.VisitComplex(&read_lambda_);
+  }
+
+  void AcceptWriteVisitor(WriteHandlingMethodVisitor<T>& v) const override
+  {
+    v.VisitComplex(&write_lambda_);
+  }
+
+  void VisitConstant(T value) override
+  {
+    read_lambda_ = [value](u32 addr) {
+      ERROR_LOG(MEMMAP, "MMIO: Read %d bytes from constant register 0x%08x -> 0x%x", sizeof(T),
```

Comment:
The reporting is just a proove of concept. If we want this in master, I have to implement a method which doesn't spam the common ERROR_LOG, at least not within the wrong group MEMMAP. This also uses %x for all kind of integers.